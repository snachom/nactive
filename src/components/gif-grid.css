.search-name, .loader {
  text-align: center;
}

.card-grid {
  display: flex;
  flex-flow: row wrap;
  /* flex-flow: column wrap; */
  width: 100%;
  justify-content: center;

}