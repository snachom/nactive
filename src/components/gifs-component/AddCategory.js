import React, { useState } from 'react';
import PropTypes from 'prop-types';

import './add-category.css';


export const AddCategory = ( {setCategories} ) => {

  const [ inputValue, setInputValue ] = useState('')

  const handleInputChange = (e) => {
    setInputValue( e.target.value )
  }

  const handleSubmit = (e) => {
    e.preventDefault();

    if( inputValue.trim().length > 2 ){
      setCategories( cats => [inputValue, ...cats])
      setInputValue('')
    }
  }

  return (
    <form className="giphy-form" onSubmit={ handleSubmit }>
      {/* <p> { inputValue } </p> */}
      <input 
        className="form-control giphy-input"
        type="text"
        value= { inputValue }
        onChange= { handleInputChange }
      />
    </form>
  )
};


AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
}
