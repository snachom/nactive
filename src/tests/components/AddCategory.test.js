import React from 'react';
import '@testing-library/jest-dom'
import {shallow} from 'enzyme'
import {AddCategory} from '../../components/AddCategory'

describe('Pruebas en <AddCategory />', () => {

  const setCategories = jest.fn() // const setCategories = () => {}
  let wrapper = shallow( <AddCategory setCategories={setCategories} />)
  
  beforeEach(() => {
    jest.clearAllMocks()
    wrapper = shallow( <AddCategory setCategories={setCategories} />) // Reset wrapper (it had value 'Hola mundo')
  })
  
  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })
  
  test('debe cambiar la caja de texto', () => {
    let input = wrapper.find('input')
    const value = 'Hola mundo'
    input.simulate('change', { target: { value: value} })
    // expect(wrapper.find('p').text().trim()).toBe(value + '1')
    // expect(wrapper.find('p').text().trim()).toBe(value)
  })

  test('no debe poster al info onSubmit', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} })
    expect(setCategories).not.toHaveBeenCalled()
  })
  
  test('debe llamar el setCategories y limpiar la caja de texto', () => {
    
    const value = 'Hi world'

    //1. simular el inputChange
      wrapper.find('input').simulate('change', {target: {value}})

    //2. simular el submit
      wrapper.find('form').simulate('submit', { preventDefault(){} })

    //3. setCategories debe haberse llamado
      expect(setCategories).toHaveBeenCalled()
      expect(setCategories).toHaveBeenCalledTimes(1)
      expect(setCategories).toHaveBeenCalledWith( expect.any(Function) )

    //4. valor de input debe estar ''
      expect(wrapper.find('input').prop('value')).toBe('')
    

  })


});
