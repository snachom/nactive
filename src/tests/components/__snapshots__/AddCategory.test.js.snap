// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory /> debe mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
